### PR TITLE
Sig cooking update

### PR DESCRIPTION
**Pull request by SigDoesCode**
## About The Pull Request
Adds meals! Meals are a new food item like custom salads intended to include main-course items like fries and steak, or eggs and waffles. To make a meal, put a food item on a plate. 

Meals can hold up to 4 items, and cannot hold other meals (get your recursive food elsewhere.) When you eat from a meal, you'll eat a bit of everything that's on the plate at a time. Meals can include custom salads too!
Here are the four stages of the meal sprite. The blobs take on the coloration of the food added, too!
image
An example of denying recursive meal creation.
image

You can either make a sane meal, like this one...
image
or an insane meal, like this one.
image

They can also have condiments added to them. 

To allow meals to be created easier, Dinnerware vendors now include 5 plates. 
image

## Why It's Good For The Game
Allows the kitchen to make more interesting stuff rather than just "here is bread, here is sandwich, here is burger."

## Changelog
```changelog
add: Meals! The kitchen is a bit more exciting now.
add: 5 plates should now be available in each Dinnerware.
```

### CODE REVIEW

**Comment by SigDoesCode** [comment]
Oh, one last thing I forgot. If the name of the plate becomes too long, it becomes a "garbage plate" like this, as a reference to the garbage plate in Rochester, NY. 
image

**Comment by hex37** [comment]
I code for a different code base when I was handed this PR to look at today. Honest to god never knew Dreammaker treated their for loops that poorly. 

I gotta go redo all my list traversals now, fuck you, also thank you.

File: zzzz_modular_occulus/code/modules/cooking/meal.dm [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/meal/attackby(obj/item/W as obj, mob/user as mob)
+
+	var/sandwich_limit = 3 //Lower limit than sandwiches because it's intended to contain more filling food.
+
+	if(src.contents.len > sandwich_limit)
+		to_chat(user, "\red The [src] plate is full!")
+		return
+	else if(istype(W, /obj/item/weapon/reagent_containers/food/snacks/meal) || istype(W, /obj/item/weapon/reagent_containers/food/snacks/csalad)) //Quick check to not make these recursive.
+		to_chat(user, "\red The [src] is too small to hold the [W]!")
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/modules/cooking/meal.dm:39]: I suggest using SPAN_WARNING here, in place of \red

File: zzzz_modular_occulus/code/modules/cooking/meal.dm [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		user.drop_from_inventory(W, src)
+		update()
+		return
+	else if(istype(W,/obj/item/weapon/reagent_containers/food/snacks))
+		to_chat(user, SPAN_NOTICE("You add [W] to \the [src]."))
+		var/obj/item/weapon/reagent_containers/F = W
+		F.reagents.trans_to_obj(src, F.reagents.total_volume)
+		user.drop_from_inventory(W, src)
+		ingredients += W
+		update_taste(W)//Occulus Edit
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/modules/cooking/meal.dm:52]: Redundant comment :p

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]

**Comment by Shadow-Quill** [comment]
Need to remove `/weapon` from atom paths

File: zzzz_modular_occulus/code/modules/cooking/meal.dm [hunk 3]
```diff
@@ -0,0 +1,101 @@
+//Most of this is the same code as csalads, but with slight changes. Mainly to numbers, partially to how things work.
+/obj/item/trash/plate
+	name = "plate"
+	desc = "A standard plate. This one is empty."
+
+/obj/item/trash/plate/attackby(obj/item/weapon/W as obj, mob/user as mob)
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/modules/cooking/meal.dm:6]:
```suggestion
/obj/item/trash/plate/attackby(obj/item/W as obj, mob/user as mob)
```

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]
Still have `/weapon/`s laying around.

**Comment by Shadow-Quill** [comment]
There are now merge conflicts

**Comment by SigDoesCode** [comment]
> There are now merge conflicts

Where? How? I'm getting absolutely nothing from the CI checker. It tells me there's a map diff render conflict, I click details, and then it just says it's unmergeable.

File: zzzz_modular_occulus/code/game/machinery/vending.dm [hunk 4]
```diff
@@ -368,6 +368,38 @@
 					/obj/item/storage/belt/utility/neotheology = 150,
 					/obj/item/storage/belt/tactical/neotheology = 150)
 
+
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/game/machinery/vending.dm:371]:
```suggestion
```

File: zzzz_modular_occulus/code/game/machinery/vending.dm [hunk 5]
```diff
@@ -368,6 +368,38 @@
...
+	/obj/item/reagent_containers/food/drinks/pitcher = 3,
+	/obj/item/reagent_containers/food/drinks/teapot = 3,
+	/obj/item/reagent_containers/food/drinks/mug = 3,
+	/obj/item/reagent_containers/food/drinks/mug/teacup = 10,
+	/obj/item/trash/plate = 5)//Adding more plates so people can make meals.
+	contraband = list(/obj/item/material/kitchen/rollingpin = 2, /obj/item/tool/knife/butch = 2)
+	auto_price = FALSE
+
+
+
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/game/machinery/vending.dm:402]:
```suggestion
```

**Review by Shadow-Quill — COMMENTED** [review]
Cleaning up whitespace

**Review by Shadow-Quill — APPROVED** [review]

**Pull request merged**